dlib/dnn/loss.h

```diff
@@ -405,6 +405,7 @@ namespace dlib
         double truth_match_iou_threshold = 0.5;
         test_box_overlap overlaps_nms = test_box_overlap(0.4);
         test_box_overlap overlaps_ignore;
+        bool scale_invariant = true;
```

Comment:
This would probably be better named something like `assume_input_layer_is_image_pyramid`

Reply:
Actually, make this a strongly typed enum.  You will also need to update the serialization code to include this field.  That's missing from this PR at the moment.

Reply:
Fixed. Didn't test serialization, though.